File: Daily-Challenges/3658_gcd_odd_even.py
def gcd(a: int, b: int) ->int:
    while b:
        r = a % b       # remainder
        a, b = b, r     # Euclidean Algorithm
    return a

class Solution:
    def gcdOfOddEvenSums(self, n: int) -> int:
        sumOdd = n ** 2
        sumEven = n * (n + 1)

        return gcd(sumOdd, sumEven)

File: Daily-Challenges/3658_gcd_odd_even.cpp
int gcd(int a, int b){
    while (b){
        int r = a % b;
        a = b;
        b = r;      // Euclidean Algorithm
    }
    return a;
}
class Solution {
public:
    int gcdOfOddEvenSums(int n) {
        int sumOdd = std::pow(n, 2);
        int sumEven = n * (n + 1);

        return gcd(sumOdd, sumEven);
    }
};